File: go/design-patterns/facade/facade/facade.go
type HomeTheaterFacade struct {
    amp       Amplifier
    lights    Lights
    projector Projector
    box       StreamingBox
}

func (h *HomeTheaterFacade) WatchMovie(title string) {
    h.lights.Dim(10)
    h.projector.On()
    h.projector.Wide()
    h.amp.On()
    h.amp.SetVolume(7)
    h.box.Play(title)
}

File: cpp/design-patterns/facade/facade/facade.cpp
class HomeTheaterFacade {
    Amplifier amp; Lights lights;
    Projector projector; StreamingBox box;
public:
    void watchMovie(const std::string& title) {
        lights.dim(10);
        projector.on();
        projector.wide();
        amp.on();
        amp.setVolume(7);
        box.play(title);
    }
};